runcontainer.go

```diff
@@ -2622,3 +2611,44 @@ func (rc *runContainer16) addOffset(x uint16) (container, container) {
 
 	return low, high
 }
+
+// intervalOverlaps returns an error if the intervals overlap e.g have non-empty intersection
+func intervalOverlaps(outer interval16, inner interval16) error {
+	if !outer.hasEmptyIntersection(inner) {
+		return errors.New("intervals overlap")
+	}
+
+	return nil
+}
+
+// validate checks the run container referential integrity
+// Ensures runs are not degenerate, non-contiguous and non-overlapping
+func (rc *runContainer16) validate() error {
+	if rc.getCardinality() == 0 {
+		return errors.New("zero intervals")
+	}
+
+	for outeridx := range rc.iv {
+
+		if rc.iv[outeridx].length == 0 {
+			return errors.New("zero run length")
+		}
+
+		for inneridx := outeridx + 1; inneridx < len(rc.iv); inneridx++ {
+			outer := rc.iv[outeridx]
+			inner := rc.iv[inneridx]
+
+			if outer.equal(inner) {
+				return errors.New("intervals were equal")
+			}
+
+			err := intervalOverlaps(outer, inner)
```

Comment:
This is not strict enough. E.g., [1,10] followed by [11,20] would be an error. There must be gaps between the runs.